x-pack/plugin/esql/compute/src/main/generated-src/org/elasticsearch/compute/data/BooleanArrayBlock.java

```diff
@@ -79,21 +98,20 @@ public BooleanBlock expand() {
             incRef();
             return this;
         }
-        // TODO use reference counting to share the vector
-        try (var builder = blockFactory().newBooleanBlockBuilder(firstValueIndexes[getPositionCount()])) {
-            for (int pos = 0; pos < getPositionCount(); pos++) {
-                if (isNull(pos)) {
-                    builder.appendNull();
-                    continue;
-                }
-                int first = getFirstValueIndex(pos);
-                int end = first + getValueCount(pos);
-                for (int i = first; i < end; i++) {
-                    builder.appendBoolean(getBoolean(i));
-                }
-            }
-            return builder.mvOrdering(MvOrdering.DEDUPLICATED_AND_SORTED_ASCENDING).build();
+        vector.incRef();
+        if (nullsMask == null) {
+            return vector.asBlock();
         }
+        BooleanArrayBlock expanded = new BooleanArrayBlock(
+            vector,
+            vector.getPositionCount(),
+            null,
+            shiftNullsToExpandedPositions(),
+            MvOrdering.DEDUPLICATED_AND_SORTED_ASCENDING,
+            blockFactory()
+        );
+        blockFactory().adjustBreaker(expanded.ramBytesUsedOnlyBlock(), true);
```

Comment:
I think we should decRef the vector if we hit the circuit breaker here.

Reply:
Oh, absolutely. Good catch, thanks!